### PR TITLE
Nix: update nixpkgs to 24.11

### PR DESCRIPTION
**Pull request by viktormalik**
The 24.11 release contains LLVM 19 so we can finally merge #3433.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

**Review by ajor — APPROVED** [review]

**Review by jordalgo — APPROVED** [review]

**Comment by jordalgo** [comment]
Hmm, not sure what all the build errors are about but those probably need fixing 😜

**Comment by viktormalik** [comment]
> Hmm, not sure what all the build errors are about but those probably need fixing 😜

Yup, working on those.

**Comment by viktormalik** [comment]
> Do we have a policy on dropping older LLVMs yet? Could drop 13 while adding 19

We don't but it's my plan to drop 13, at least from the CI. Supporting last 6 versions (i.e. 3 years back) sounds reasonable to me.

Btw, I'm working on putting down a release policy/schedule so this could be a part of it.

**Comment by viktormalik** [comment]
It took a bit more work to get the CI pass so I'm re-requesting the reviews. See the first 3 commits.

**Review by jordalgo — APPROVED** [review]
LGTM.

One minor nit: Not sure if there should be new semantic_analyser tests for the added error cases where `get_int_literal` returns a nullopt.

**Comment by viktormalik** [comment]
> One minor nit: Not sure if there should be new semantic_analyser tests for the added error cases where `get_int_literal` returns a nullopt.

That should never happen as the same is checked by `check_arg`. So, this new condition is just to silence the warning (and cover cases when we change/misuse `check_arg`).

**Comment by jordalgo** [comment]
> That should never happen as the same is checked by check_arg. So, this new condition is just to silence the warning (and cover cases when we change/misuse check_arg).

Ah ok, so then should you use `LOG(BUG)` instead?

**Comment by viktormalik** [comment]
> Ah ok, so then should you use `LOG(BUG)` instead?

TBH, I don't know. I think both ways are fine as technically it's just double checking the same semantic error. But if people prefer `LOG(BUG)`, I can change it.

**Comment by jordalgo** [comment]
> TBH, I don't know. I think both ways are fine as technically it's just double checking the same semantic error. But if people prefer LOG(BUG), I can change it.

If users are not supposed to see the error then it should be `LOG(BUG)`. https://github.com/bpftrace/bpftrace/blob/master/docs/coding_guidelines.md#log-messages
But if it's a gray area then this shouldn't block.

File: src/bpffeature.cpp [hunk 1]
```diff
@@ -513,7 +513,7 @@ bool BPFfeature::has_uprobe_multi()
 #else
   has_uprobe_multi_ = false;
 #endif // HAVE_LIBBPF_UPROBE_MULTI
-  return *has_uprobe_multi_;
+  return *has_uprobe_multi_; // NOLINT
```
Comment (danobi) [inline src/bpffeature.cpp:516]: Nit: specify the lint:

```
// NOLINT(bugprone-unchecked-optional-access)
```

**Comment by danobi** [comment]
Yeah if it's a double check I'd lean towards LOG(BUG). Makes the intent more clear that it should've already been checked before reaching the double check.

File: src/bpffeature.cpp (hunk 1 above)
Comment (viktormalik) [reply src/bpffeature.cpp]: Good idea, thanks.

**Comment by viktormalik** [comment]
Ok, changed to `LOG(BUG)`.

File: src/ast/passes/semantic_analyser.cpp [hunk 2]
```diff
@@ -1361,7 +1365,9 @@ void SemanticAnalyser::visit(Call &call)
       }
     } else if (arg.type.IsIntTy() && arg.is_literal) {
       auto sig = bpftrace_.get_int_literal(&arg);
-      if (!sig.has_value() || *sig < 1 || *sig > 64) {
+      if (!sig.has_value())
+        LOG(ERROR, call.loc, err_) << call.func << ": invalid signal value";
```
Comment (viktormalik) [inline src/ast/passes/semantic_analyser.cpp:1369]: I had to go with `LOG(ERROR)` here b/c while the above `get_int_literal` does `LOG(ERROR)`, it doesn't print to `err_` so semantic analyser doesn't properly fail.

**Review by danobi — APPROVED** [review]

**Pull request merged**